cli/stage_parser.py

```diff
@@ -35,6 +36,50 @@ class OrchestrationNotFound(Exception):
     pass
 
 
+class RenderingException(Exception):
+    """
+    Exception class that represents a rendering error
+    """
+    def __init__(self, error_desc):
+        super(RenderingException, self).__init__()
+        self.error_desc = error_desc
+
+    def pretty_error_desc_str(self):
+        """
+        Returns a more user-friendly message of the exception
+        """
+        output = ""
+        if self.error_desc.find("SaltRenderError") == -1:
+            found_ex_line = True
```

Comment:
I don't understand this piece. That reads to me as if it would set found_ex_line = True if "SaltRenderError" was *not* found? Then later, it checks if each line starts with "SaltRenderError" anyway?

Reply:
The idea here is to remove the stack trace part of a `SaltRenderError` exception and keep the rest of the exception output, but keep the stack trace part of any other exception that might be caught.

I agree that the way this is written is confusing, I'll change this into a more readable piece of code.